### PR TITLE
Additional Tests and Test Support Improvements

### PR DESCRIPTION
**Pull request by mgacy**
This is focused on extending test coverage and cleaning up test support.

- gather code coverage
- add additional `ImageFetcher` tests (91 % coverage)
- add `ImageProcessor` tests (93% coverage)
- split `Helpers.swift` file into separate files in `Tests/ImageFetcherTests/Helpers`; I felt it was getting to be too much of a hodgepodge of different types and extensions
- call `XCTFail` in the default method implementations of `MockCache` and `Networking.mock`
- remove performance tests: I was using `XCTestCase.measureMetrics()` for much larger areas of code than it is intended for

I also threw in a fix for an incorrect documentation reference

This concludes my intended changes

### CODE REVIEW

File: Tests/ImageFetcherTests/Helpers/Extensions/Color+Utils.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
+//
+//  Color+Utils.swift
```
Comment (mgacy) [inline Tests/ImageFetcherTests/Helpers/Extensions/Color+Utils.swift:2]: No changes here; just a separate file

File: Tests/ImageFetcherTests/Helpers/Extensions/XCTestCase+Utils.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
+//
+//  XCTestCase+Utils.swift
```
Comment (mgacy) [inline Tests/ImageFetcherTests/Helpers/Extensions/XCTestCase+Utils.swift:2]: No changes here; just a separate file

File: Tests/ImageFetcherTests/Helpers/Mock.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
+//
+//  Mock.swift
```
Comment (mgacy) [inline Tests/ImageFetcherTests/Helpers/Mock.swift:2]: No changes here; just a separate file

File: Tests/ImageFetcherTests/Helpers/MockImageProcessor.swift [hunk 4]
```diff
@@ -0,0 +1,71 @@
+//
+//  MockImageProcessor.swift
```
Comment (mgacy) [inline Tests/ImageFetcherTests/Helpers/MockImageProcessor.swift:2]: No changes here; just a separate file

**Review by jagreenwood — APPROVED** [review]

**Pull request merged**